drivers/builtin/src/asn1parse.c

```diff
@@ -179,6 +179,45 @@ int mbedtls_asn1_get_mpi(unsigned char **p,
 }
 #endif /* MBEDTLS_BIGNUM_C */
 
+int mbedtls_asn1_get_integer(unsigned char **p, const unsigned char *end,
+                             unsigned char **head, size_t *length)
+{
+    int ret;
+    size_t integer_length;
+    unsigned char *start = *p;
+
+    *length = 0;
+    *head = NULL;
+
+    if ((ret = mbedtls_asn1_get_tag(p, end, &integer_length, MBEDTLS_ASN1_INTEGER)) != 0) {
```

Comment:
Note that at this point, `integer_length` is the length of the content, not the length of the value. For a valid input, the content can include a leading 0 byte (when the value's most significant bit is at position 8*n+7).

This function has a bug: it doesn't update `head` and `integer_length` to remove the leading zero byte. A more precise name (e.g. `content_length` for the content, `value_length` for the part that contains actual value octets) would make this more visible.